src/core/qgsziputils.cpp

```diff
@@ -84,8 +85,19 @@ bool QgsZipUtils::unzip( const QString &zipFilename, const QString &dir, QString
         char *buf = new char[len];
         if ( zip_fread( file, buf, len ) != -1 )
         {
-          QFileInfo newFile( QDir( dir ), QString( stat.name ) );
-          std::ofstream( newFile.absoluteFilePath().toStdString() ).write( buf, len );
+          QString fileName( stat.name );
+          QFileInfo newFile( QDir( dir ), fileName );
+          bool isDirectory = fileName.endsWith( '/' );
```

Comment:
Is this windows compatible?

Reply:
cannot http://doc.qt.io/qt-5/qfileinfo.html#isDir be used?

Reply:
Sounds like a good plan.